broker/src/main/java/io/camunda/zeebe/broker/system/partitions/impl/steps/StreamProcessorPartitionTransitionStep.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.broker.system.partitions.impl.steps;
+
+import io.camunda.zeebe.broker.system.partitions.PartitionTransitionContext;
+import io.camunda.zeebe.broker.system.partitions.PartitionTransitionStep;
+import io.camunda.zeebe.protocol.record.PartitionRole;
+import io.camunda.zeebe.util.sched.future.ActorFuture;
+import io.camunda.zeebe.util.sched.future.CompletableActorFuture;
+
+public class StreamProcessorPartitionTransitionStep implements PartitionTransitionStep {
+
+  @Override
+  public ActorFuture<PartitionTransitionContext> transitionTo(
+      final PartitionTransitionContext context,
+      final long currentTerm,
+      final PartitionRole currentRole,
+      final long nextTerm,
+      final PartitionRole targetRole) {
+
+    final var streamProcessor = context.getStreamProcessor();
+
+    switch (targetRole) {
+      case LEADER:
+        {
+          if (!context.shouldProcess()) {
+            streamProcessor.pauseProcessing();
+          } else {
+            streamProcessor.resumeProcessing();
+          }
```

Comment:
Hm this will not work. We dont want to resume where we left of when we become leader again. I dont know whether we can solve it like that, especially when we execute the transition steps in parallel. There are some prerequiste things we have to do. But maybe we can combine them together?